### PR TITLE
[config_mgmt.py]: Separate class for Dy Port BreakOut and support ext…

### PR DESCRIPTION
**Pull request by praveen-li**
…ra config tables listing.

Changes:
1.) Added a new class for ConfigMgmtDPB, which is derived from ConfigMgmt.
2.) Support to list extra tables in config.
3.) Search keys in config. This is need to list port related config in extra tables.
4.) Separate loadData function in ConfigMgmt class.

Signed-off-by: Praveen Chaudhary pchaudhary@linkedin.com

**- What I did**

1.) Added a new class for ConfigMgmtDPB, which is derived from ConfigMgmt.
2.) Support to list extra tables in config.
3.) Search keys in config. This is need to list port related config in extra tables.
4.) Separate loadData function in ConfigMgmt class.

**- How I did it**
0.) https://github.com/zhenggen-xu/sonic-buildimage/pull/69 to store extra tables.
1.) Added a new class for ConfigMgmtDPB, which is derived from ConfigMgmt.
2.) Support to list extra tables in config.
3.) Search keys in config. This is need to list port related config in extra tables.
4.) Separate loadData function in ConfigMgmt class.

**- How to verify it**
Ran below Tests 
```
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 4x25G 
Target Breakout Mode : 2x50G

Ports to be deleted : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}
Ports to be added : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
} 
Final list of ports to be added :  
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "4x25G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet2
Find dependecies for port Ethernet3
Find dependecies for port Ethernet0
Find dependecies for port Ethernet1
Deleting Port: Ethernet2
Deleting Port: Ethernet3
Deleting Port: Ethernet0
Deleting Port: Ethernet1
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Generating default config for ['Ethernet2', 'Ethernet0']
Merge Default Config for ['Ethernet2', 'Ethernet0']
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet0  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet2  | untagged       |                       |
|           | fe80::1/10             | Ethernet4  | untagged       |                       |
|           |                        | Ethernet5  | untagged       |                       |
|           |                        | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       xe0( 68)  down   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe1( 69)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe2( 70)  down   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe3( 71)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe4( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe6( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 2x50G 
Target Breakout Mode : 1x100G

Ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}
Ports to be added : 
 {
    "Ethernet0": "100000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
} 
Final list of ports to be added :  
 {
    "Ethernet0": "100000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "2x50G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet2
Find dependecies for port Ethernet0
Dependecies Exist. No further action will be taken
*** Printing dependecies ***
/sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[vlan_name='Vlan100']/members[.='Ethernet2']
/sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[vlan_name='Vlan777']/members[.='Ethernet2']
/sonic-vlan:sonic-vlan/VLAN_MEMBER/VLAN_MEMBER_LIST[vlan_name='Vlan100'][port='Ethernet2']/port
/sonic-vlan:sonic-vlan/VLAN_MEMBER/VLAN_MEMBER_LIST[vlan_name='Vlan777'][port='Ethernet2']/port
/sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[vlan_name='Vlan100']/members[.='Ethernet0']
/sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[vlan_name='Vlan777']/members[.='Ethernet0']
/sonic-vlan:sonic-vlan/VLAN_MEMBER/VLAN_MEMBER_LIST[vlan_name='Vlan100'][port='Ethernet0']/port
/sonic-vlan:sonic-vlan/VLAN_MEMBER/VLAN_MEMBER_LIST[vlan_name='Vlan777'][port='Ethernet0']/port
:::Lock PID: None and self.pid:29100:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet0  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet2  | untagged       |                       |
|           | fe80::1/10             | Ethernet4  | untagged       |                       |
|           |                        | Ethernet5  | untagged       |                       |
|           |                        | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       xe0( 68)  !ena   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe1( 69)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe2( 70)  !ena   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe3( 71)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe4( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe6( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 2x50G 
Target Breakout Mode : 1x100G

Ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}
Ports to be added : 
 {
    "Ethernet0": "100000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
} 
Final list of ports to be added :  
 {
    "Ethernet0": "100000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "2x50G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet2
Find dependecies for port Ethernet0
Deleting Port: Ethernet2
Deleting Port: Ethernet0
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet4  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet5  | untagged       |                       |
|           | fe80::1/10             | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
|           |                        | Ethernet9  | untagged       |                       |
|           |                        | Ethernet10 | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       ce0( 68)  down   4  100G  FD   SW  No   Forward          None    D    KR4  9412    No      
       xe0( 69)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe1( 70)  !ena   2     -       SW  No   Forward          None    D   None  9412    No      
       xe2( 71)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe3( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe4( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 1x100G 
Target Breakout Mode : 4x25G

Ports to be deleted : 
 {
    "Ethernet0": "100000"
}
Ports to be added : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet0": "100000"
} 
Final list of ports to be added :  
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "1x100G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet0
Deleting Port: Ethernet0
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet4  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet5  | untagged       |                       |
|           | fe80::1/10             | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
|           |                        | Ethernet9  | untagged       |                       |
|           |                        | Ethernet10 | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       xe0( 68)  down   1   25G  FD   SW  No   Forward          None    D     KR  9412    No      
       xe1( 69)  down   1   25G  FD   SW  No   Forward          None    D     KR  9412    No      
       xe2( 70)  down   1   25G  FD   SW  No   Forward          None    D     KR  9412    No      
       xe3( 71)  down   1   25G  FD   SW  No   Forward          None    D     KR  9412    No      
       xe4( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe6( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 4x25G 
Target Breakout Mode : 1x100G

Ports to be deleted : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}
Ports to be added : 
 {
    "Ethernet0": "100000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
} 
Final list of ports to be added :  
 {
    "Ethernet0": "100000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "4x25G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet2
Find dependecies for port Ethernet3
Find dependecies for port Ethernet0
Find dependecies for port Ethernet1
Deleting Port: Ethernet2
Deleting Port: Ethernet3
Deleting Port: Ethernet0
Deleting Port: Ethernet1
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet4  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet5  | untagged       |                       |
|           | fe80::1/10             | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
|           |                        | Ethernet9  | untagged       |                       |
|           |                        | Ethernet10 | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       ce0( 68)  down   4  100G  FD   SW  No   Forward          None    D    KR4  9412    No      
       xe0( 69)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe1( 70)  !ena   2     -       SW  No   Forward          None    D   None  9412    No      
       xe2( 71)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe3( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe4( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 1x100G 
Target Breakout Mode : 2x50G

Ports to be deleted : 
 {
    "Ethernet0": "100000"
}
Ports to be added : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet0": "100000"
} 
Final list of ports to be added :  
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "1x100G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet0
Deleting Port: Ethernet0
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Generating default config for ['Ethernet2', 'Ethernet0']
Merge Default Config for ['Ethernet2', 'Ethernet0']
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet0  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet2  | untagged       |                       |
|           | fe80::1/10             | Ethernet4  | untagged       |                       |
|           |                        | Ethernet5  | untagged       |                       |
|           |                        | Ethernet6  | untagged       |                       |
|           |                        | Ethernet7  | untagged       |                       |
|           |                        | Ethernet8  | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       xe0( 68)  down   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe1( 69)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe2( 70)  down   2   50G  FD   SW  No   Forward          None   FA    KR2  9412    No      
       xe3( 71)  !ena   1     -       SW  No   Forward          None    D   None  9412    No      
       xe4( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe6( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
:::Lock Acquired:::
:::Lock Timer Extended:::

Running Breakout Mode : 2x50G 
Target Breakout Mode : 4x25G

Ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
}
Ports to be added : 
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}

After running Logic to limit the impact

Final list of ports to be deleted : 
 {
    "Ethernet2": "50000", 
    "Ethernet0": "50000"
} 
Final list of ports to be added :  
 {
    "Ethernet2": "25000", 
    "Ethernet3": "25000", 
    "Ethernet0": "25000", 
    "Ethernet1": "25000"
}
Loaded below Yang Models
['sonic-acl', 'sonic-extension', 'sonic-head', 'sonic-interface', 'sonic-loopback-interface', 'sonic-port', 'sonic-portchannel', 'sonic-vlan']
Reading data from Redis configDb
Note: Below table(s) have no YANG models:
VERSIONS, DEVICE_METADATA, DEVICE_NEIGHBOR, LOCK, FLEX_COUNTER_TABLE, BREAKOUT_CFG, 
Below Config can not be verified, It may cause harm to the system
 {
  "BREAKOUT_CFG": {
    "Ethernet0": {
      "brkout_mode": "2x50G"
    }
  }
}
Do you wish to Continue? [y/N]: 
Start Port Deletion
Find dependecies for port Ethernet2
Find dependecies for port Ethernet0
Deleting Port: Ethernet2
Deleting Port: Ethernet0
Data Validation successful
Generate Final Config to write in DB

Start Port Addition
Generating default config for ['Ethernet2', 'Ethernet3', 'Ethernet0', 'Ethernet1']
Merge Default Config for ['Ethernet2', 'Ethernet3', 'Ethernet0', 'Ethernet1']
Data Validation successful
Generate Final Config to write in DB
Writing in Config DB
Verify Port Deletion from Asic DB, Wait...
Writing in Config DB
Breakout process got successfully completed.
:::Lock Released:::
+-----------+------------------------+------------+----------------+-----------------------+
|   VLAN ID | IP Address             | Ports      | Port Tagging   | DHCP Helper Address   |
+===========+========================+============+================+=======================+
|       100 | 2a04:f547:43:6a0::1/64 | Ethernet0  | untagged       | 10.186.72.116         |
|           | 10.187.40.1/26         | Ethernet1  | untagged       |                       |
|           | fe80::1/10             | Ethernet2  | untagged       |                       |
|           |                        | Ethernet3  | untagged       |                       |
|           |                        | Ethernet4  | untagged       |                       |
|           |                        | Ethernet5  | untagged       |                       |
|           |                        | Ethernet6  | untagged       |                       |
ps

                 ena/        speed/ link auto    STP                  lrn  inter   max   cut   loop
           port  link  Lns   duplex scan neg?   state   pause  discrd ops   face frame  thru?  back
       xe0( 68)  down   1   25G  FD   SW  No   Forward          None   FA     KR  9412    No      
       xe1( 69)  down   1   25G  FD   SW  No   Forward          None   FA     KR  9412    No      
       xe2( 70)  down   1   25G  FD   SW  No   Forward          None   FA     KR  9412    No      
       xe3( 71)  down   1   25G  FD   SW  No   Forward          None   FA     KR  9412    No      
       xe4( 72)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe5( 73)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      
       xe6( 74)  !ena   1   25G  FD   SW  No   Forward          None   FA     KR  9122    No      

```

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: config/config_mgmt.py
Comment (vasant17) [inline config/config_mgmt.py]: indentation

File: config/config_mgmt.py
Comment (vasant17) [inline config/config_mgmt.py:526]: why don't we prepare these patterns ahead of time using skeys and then search key in prepared patterns

Comment (praveen-li) [reply config/config_mgmt.py:526]: @vasant17 : any advantage of it?

One thing I want, is to keep these patterns at only one place, may be here or at start of library. So that if in future, configDB key pattern changes, we change at one place, except that Let me know if you see any other constraints.

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py]: only line 95 right? will fix it.

File: config/config_mgmt.py
Comment (vasant17) [inline config/config_mgmt.py:539]: if Out.get(key) is None, doesn't it mean we did NOT find key in any skeys? If so, what are we doing here by recursively calling the function? Could you please explain.

File: config/config_mgmt.py
Comment (vasant17) [inline config/config_mgmt.py:541]: I don't see searchKeysInConfig method returning anything. Is it supposed to return "found"?

File: config/config_mgmt.py
Comment (vasant17) [inline config/config_mgmt.py:578]: This should have been more generic and should be used to get default config for any object. But we can think of it in later versions

File: config/config_mgmt.py
Comment (vasant17) [reply config/config_mgmt.py:526]: I was thinking of the same thing. If the skeys are static and we have superset, we can create all patterns only once through out the life of the program. Ans we can just update/change it if any new key gets introduced in config DB as we bring on new feature. 

If that is NOT the case and we want to prepare pattern for every skey in this method, I was trying to avoid creating/constructing pattern multiple times for the same skey. Hence I suggested to do it in the beginning and then iterate over keys to search in patterns.

File: config/config_mgmt.py
Comment (vasant17) [reply config/config_mgmt.py]: yes

File: config/config_mgmt.py
Comment (vasant17) [reply config/config_mgmt.py:541]: > I don't see searchKeysInConfig method returning anything. Is it supposed to return "found"?

Sorry, I see it below that we are returning "found"

**Review by vasant17 — APPROVED** [review]

**Comment by zhenggen-xu** [comment]
@praveen-li @vasant17 Please resolve all the conversations before we approve/merge the PR.

File: config/config_mgmt.py
Comment (zhenggen-xu) [inline config/config_mgmt.py]: Why not calling "loadData" directly to reuse code?

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py:578]: Under ConfigMgmtDPB class we do not want any other default config then related to the ports, so better to keep as ports as of now.

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py:539]: Here In Dict will be somethign like below:
```
"LOOPBACK_INTERFACE": {
			"Loopback0": {},
			"Loopback0|2a04:5555:40:4::4e9/128": {
				"scope": "global",
				"family": "IPv6"
			},
			"Loopback0|10.184.8.233/32": {
				"scope": "global",
				"family": "IPv4"
			}
		},
		"CRM": {
			"Config": {
				"polling_interval": "0"
			}
		}
"DEVICE_NEIGHBOR": {
			"Ethernet112": {
				"name": "dccsw01.nw",
				"port": "Eth18"
			},
			"Ethernet114": {
				"name": "dccsw02.nw",
				"port": "Eth18"
			},
			"Ethernet116": {
				"name": "dccsw03.nw",
				"port": "Eth18"
			},
			"Ethernet118": {
				"name": "dccsw04.nw",
				"port": "Eth18"
			}
		},

```
and Out dict must be 
```
DEVICE_NEIGHBOR": {
			"Ethernet114": {
				"name": "dccsw02.nw",
				"port": "Eth18"
			}
}
```
if skeys is [Ethermet114].

**Comment by zhenggen-xu** [comment]
@praveen-li We need add Yang models for BREAKOUT_CFG table,  or we skip this table check in ConfigMgmtDPB class,  so to avoid the confirmation ask as we know that is expected.

**Comment by praveen-li** [comment]
Sure, I need to resolve couple more conversations by one more commit.


Thanks,

Regards
Praveen
________________________________

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py:526]: The idea is creating pattern once is good, but I will solve it soon after I finish DPB system tests so that I do not have a retest this entire critical part manually.

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py]: Done.

File: config/config_mgmt.py
Comment (praveen-li) [reply config/config_mgmt.py]: yes this can be called, since I am already calling other class functions in constructor.

Comment (praveen-li) [reply config/config_mgmt.py]: done.

**Comment by praveen-li** [comment]
@vasant17 @zhenggen-xu  Conflict resolved, Kindly merge. I do not have merge permissions for this.

**Pull request merged**
